Soft Deleted User Restore

sanjacornelius opened the pull request:
This PR adds the ability to restore soft-deleted users.

**To Test:**

- Create a user
- Delete a user
- Try to recreate the same user (same username and or same email)

**Result:**
A modal appears that indicates a deleted user exists with the email address or username and asks if they would like to restore the user.

If they select 'save', then restore the user.

velkymx commented on resources/views/admin/users/index.blade.php:
```diff
@@ -196,6 +228,17 @@
+                            restoreUserModal.usernameToRestore = restoreUserModal.restoreValue= this.username;
+                            restoreUserModal.restoreAttribute = 'username';
+                        }
+                        $('#addUser').modal('hide');
```
Is this jQuery?!? Please update to use VueJS https://bootstrap-vue.js.org/docs/components/modal/

sanjacornelius commented:
@velkymx 

**Changes:**
- Created a separate Vue component for the Add User Modal
- Converted the Add User Modal to Vue-Bootstrap modal
- Removed Restore User Modal and replaced it with Vue-Bootstrap Message Box.
